cofmpy/data_stream_handler/kafka_utils.py

```diff
@@ -0,0 +1,182 @@
+            raise ValueError(f"Timeout must be non-negative, got {self.timeout}")
+
+        valid_offsets = {"earliest", "latest", "none"}
+        if self.auto_offset_reset not in valid_offsets:
+            raise ValueError(
+                f"Invalid auto_offset_reset: '{self.auto_offset_reset}'. "
+                f"Must be one of {valid_offsets}"
+            )
+
+        interpolator = Interpolator()
+        valid_interpolation = set(interpolator._registry.keys())
+        if self.interpolation not in valid_interpolation:
+            raise ValueError(
+                f"Invalid interpolation method: '{self.interpolation}'. "
+                f"Must be one of {valid_interpolation}"
+            )
+
+    @classmethod
+    def from_dict(cls, config: Dict[str, Any]) -> "KafkaHandlerConfig":
+        """Builds up the configuration Dataclass
+
+        Args:
+            config (dict): kwargs-like dictionary
+
+        Returns:
+            dataclass: class storing config data
+        """
+
+        required = {"uri", "topic", "variable", "group_id"}
+        missing = required - set(config.keys())
+        if missing:
+            for key in missing:
+                logger.error(f"Missing kafka config field: {key}")
+            raise KeyError("Configuration for KafkaDataStreamHandler is incomplete")
+
+        try:
+            server_url, port = config["uri"].split(":")
+        except ValueError as exc:
+            raise ValueError(
+                f"Malformed URI in Kafka config: {config['uri']!r}"
+            ) from exc
+
+        return cls(
+            topic=config["topic"],
+            variable=config["variable"],
+            server_url=server_url,
+            port=port,
+            group_id=config["group_id"],
+            timeout=config.get("timeout", cls.timeout),
+            interpolation=config.get("interpolation", cls.interpolation),
+            auto_offset_reset=config.get("auto_offset_reset", cls.auto_offset_reset),
+            enable_auto_commit=config.get("enable_auto_commit", cls.enable_auto_commit),
+        )
+
+
+class KafkaThreadManager:
+    """Class managing the thread for the kafka consumer"""
+
+    def __init__(self, consumer, callback, thread_lifetime=40):
+        """Constructor requires the confluent_kafka consumer object
+        and on_consume callback
+
+        Args:
+            consumer (_type_): _description_
+            callback (function): _description_
+        """
+        self.consumer = consumer
+        self.callback = callback
+        self.thread_lifetime = thread_lifetime
+        self.running = False
+        self.thread = None
+        self.start_time = None
+
+    def start(self):
+        """Start consuming thread"""
+        if self.thread_lifetime:
+            self.start_time = time.time()
+        if not self.running:
+            self.running = True
+            self.thread = threading.Thread(target=self._consume_loop)
+            self.thread.daemon = True
```

Comment:
As we are not very familiar with all this stuff, maybe you could add some short description to help us. For example 
```
self.thread.daemon = True  # Do this blabla
```